dipy/reconst/dti.py

```diff
@@ -808,21 +831,32 @@ def predict(self, dti_params, S0=1.):
 
 class TensorFit(object):
 
-    def __init__(self, model, model_params):
+    def __init__(self, model, model_params, model_S0=None):
         """ Initialize a TensorFit class instance.
         """
         self.model = model
         self.model_params = model_params
+        self.model_S0 = model_S0
 
     def __getitem__(self, index):
         model_params = self.model_params
+        model_S0 = self.model_S0
         N = model_params.ndim
         if type(index) is not tuple:
             index = (index,)
         elif len(index) >= model_params.ndim:
             raise IndexError("IndexError: invalid index")
         index = index + (slice(None),) * (N - len(index))
-        return type(self)(self.model, model_params[index])
+        if model_S0 is None:
```

Comment:
But `model_S0=model_S0[index_S0]` would fail if `model_S0` is `None`. Or are you suggesting just pass it along if it's `None` anyway on line 848?